crypto/ecdsa/public.go

```diff
@@ -0,0 +1,78 @@
+package neofsecdsa
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/sha256"
+	"crypto/sha512"
+	"math/big"
+
+	"github.com/nspcc-dev/neo-go/pkg/crypto/keys"
+)
+
+// PublicKey implements neofscrypto.PublicKey based on ECDSA.
+//
+// Supported schemes:
+//  - neofscrypto.ECDSA_SHA512 (default)
+//  - neofscrypto.ECDSA_DETERMINISTIC_SHA256
+type PublicKey struct {
+	deterministic bool
+
+	key ecdsa.PublicKey
+}
+
+// MakeDeterministic makes PublicKey to use Deterministic ECDSA scheme
+// (see neofscrypto.ECDSA_DETERMINISTIC_SHA256). By default,
+// neofscrypto.ECDSA_SHA512 is used.
+func (x *PublicKey) MakeDeterministic() {
+	x.deterministic = true
+}
+
+// Decode decodes compressed binary representation of the PublicKey.
+//
+// See also Signer.EncodePublicKey.
+func (x *PublicKey) Decode(data []byte) error {
+	pub, err := keys.NewPublicKeyFromBytes(data, elliptic.P256())
```

Comment:
`NewPublicKeyFromBytes` also decodes decompressed representations, could you mention, that we allow them?

Reply:
Yes, this should be definitely reflected in docs.